app/models/setting/provisioning.rb

```diff
@@ -1,5 +1,11 @@
 require 'facter'
 class Setting::Provisioning < Setting
+  GENERATOR_TYPES = {
+    'Off' => _('Off'),
+    'Random-based' => _('Random-based'),
+    'MAC-based' => _('MAC-based')
+  }
```

Comment:
I think, it would be cleaner to store this in the name_generator service. You can then call `NameGenerator::Types` below.